src/Symfony/Component/Config/Tests/Builder/GeneratedConfigTest.php

```diff
@@ -159,10 +159,12 @@ public function testSetExtraKeyMethodIsNotGeneratedWhenAllowExtraKeysIsFalse()
      */
     private function generateConfigBuilder(string $configurationClass, ?string &$outputDir = null)
     {
-        $outputDir = tempnam(sys_get_temp_dir(), 'sf_config_builder_');
-        unlink($outputDir);
-        mkdir($outputDir);
-        $this->tempDir[] = $outputDir;
+        if (null === $outputDir) {
```

Comment:
$outputDir is going to be always null
what we need is probably

```suggestion
        if (1 < \func_num_args()) {
```

Reply:
Lines 87 -> 89 of the test contain commented code, that when uncommented try to regenerate the snapshot files.
On line 88 it specifically passes the directory of the expected code, so the new snapshots are generated in the given location. Resulting in the outputDir not being null.

Reply:
got it thanks